fitbenchmarking/results_processing/tables.py

```diff
@@ -64,62 +67,68 @@ def create_results_tables(options, results, best_results, group_name,
         if suffix in options.table_type:
-            table_names[suffix] = \
-                '{0}_{1}_{2}_table.'.format(group_name,
-                                            suffix,
-                                            options.cost_func_type)
-
-            table, html_table, txt_table = generate_table(results,
-                                                          best_results,
-                                                          options,
-                                                          group_dir,
-                                                          pp_locations,
-                                                          table_names[suffix],
-                                                          suffix)
-
-            table_title = table.table_title
-            file_path = table.file_path
-
-            description = table.get_description(description)
-
-            table_format = None if suffix == 'local_min' \
-                else description[options.comparison_mode]
-
-            has_pp = table.has_pp
-
-            pp_filenames = table.pp_filenames
-
-            root = os.path.dirname(getfile(fitbenchmarking))
-            template_dir = os.path.join(root, 'templates')
-            css = get_css(options, group_dir)
-            js = get_js(options, group_dir)
-            env = Environment(loader=FileSystemLoader(template_dir))
-            template = env.get_template("table_template.html")
-            html_output_file = file_path + 'html'
-            txt_output_file = file_path + 'txt'
-
-            with open(txt_output_file, "w") as f:
-                f.write(txt_table)
-            failed_minimzers = sum(list(unselected_minimzers.values()), [])
-            report_failed_min = failed_minimzers != []
-
-            with open(html_output_file, "w", encoding="utf-8") as f:
-                f.write(
-                    template.render(css_style_sheet=css['main'],
-                                    custom_style=css['custom'],
-                                    table_style=css['table'],
-                                    mathjax=js['mathjax'],
-                                    table_description=description[suffix],
-                                    table_format=table_format,
-                                    result_name=table_title,
-                                    has_pp=has_pp,
-                                    pp_filenames=pp_filenames,
-                                    table=html_table,
-                                    error_message=ERROR_OPTIONS,
-                                    failed_problems=failed_problems,
-                                    unselected_minimzers=unselected_minimzers,
-                                    algorithm_type=options.algorithm_type,
-                                    report_failed_min=report_failed_min))
+
+            if suffix == 'local_min' and options.cost_func_type == 'poisson':
```

Comment:
This is a bit too specific. If we add a new cost function which is not a least squares one in the future, then we'll have to extend this with each new cost function.

I think this should be handled in the class that creates the table, either by exceptions or a validate function.

E.g. the local_min table class raises an exception if it's called with results from a cost function which is not a subclass of BaseNLLSCostFunc (`not isinstance(..., BaseNLLSCostFunc)`). We could then catch the exception in a try except around `generate_table` with a slightly more generic warning and `continue` to skip to the next table type.